pac4j-saml-config-db/src/main/java/org/pac4j/saml/dbclient/DatabaseSAML2ClientConfiguration.java

```diff
@@ -0,0 +1,225 @@
+/*
+  Copyright 2012 - 2015 pac4j organization
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package org.pac4j.saml.dbclient;
+
+import org.apache.commons.lang3.StringUtils;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.saml.client.AbstractSAML2ClientConfiguration;
+import org.pac4j.saml.client.SAML2ClientConfiguration;
+import org.pac4j.saml.dbclient.dao.api.DbLoadedSamlClientConfigurationDto;
+import org.pac4j.saml.dbclient.dao.api.SamlClientDao;
+
+
+/**
+ * SAML2 Client configuration intended to be loaded from a database.
+ * 
+ * An alternative to {@link SAML2ClientConfiguration}. Unlike {@link SAML2ClientConfiguration}, this class does not use paths to resources.
+ * Instead, it contains the data directly.
+ *
+ * @author jkacer
+ * @since 1.9.0
+ */
+public class DatabaseSAML2ClientConfiguration extends AbstractSAML2ClientConfiguration {
+
+	/** DAO to read SAML client configurations. Should be shared by all configuration instances. */
+	private final SamlClientDao dao;
+	
+	/** Binary data of a JKS keystore. */
+	private byte[] keystoreBinaryData;
+
+    /** IdP metadata. */
+    private String identityProviderMetadata;
+
+	/** SAML Client name. */
+	private String clientName;
+    
+    // ------------------------------------------------------------------------------------------------------------------------------------
+    
+    
+    /**
+	 * The constructor. Use setters to initialize properties.
+	 * 
+	 * @param dao
+	 *            DAO loading client configuration from DB.
+	 */
+    public DatabaseSAML2ClientConfiguration(final SamlClientDao dao) {
+    	super();
+    	if (dao == null) {
+    		throw new IllegalArgumentException("DAO must not be null.");
+    	}
+    	this.dao = dao;
+    }
+    
+
+	public byte[] getKeystoreBinaryData() {
+		return keystoreBinaryData;
+	}
+
+	public void setKeystoreBinaryData(byte[] keystoreBinaryData) {
+		this.keystoreBinaryData = keystoreBinaryData;
+	}
+
+	public String getIdentityProviderMetadata() {
+		return identityProviderMetadata;
+	}
+
+	public void setIdentityProviderMetadata(String identityProviderMetadata) {
+		this.identityProviderMetadata = identityProviderMetadata;
+	}
+
+	
+	/* (non-Javadoc)

```

Comment:
These 4 methods are setter/getters for keystoreBinaryData and identityProviderMetadata. These two attributes are returned by getResolvedKeystoreData() and getResolvedIdentityProviderMetadata() - called in SAML2Client.

You are right that these setters/getters are not called anywhere. I just put them there to have setters/getters for the fields but I can remove them without any impact.

They are not annotated with @Override because they are introduced in this class, not a parent class.